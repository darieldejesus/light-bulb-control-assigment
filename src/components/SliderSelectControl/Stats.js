import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import SunIcon from './SunIcon';

const Container = styled.div`
font-size: 30px;
  color: white;
  left: 0;
  pointer-events: none;
  position: absolute;
  right: 0;
  text-align: center;
  top: 35%;
`;

const PercentageLabel = styled.span`
  position: relative;
  color: white;
  display: block;
  font-size: 1em;
  margin: 0.2em auto;
  text-align: center;

  span {
    bottom: 15%;
    font-size: 0.4em;
    margin-left: 0.1em;
    position: absolute;
  }
`;

const BrightnessLabel = styled.p`
  color: #adadad;
  margin: 0;
  font-size: 0.4em;
`;

const Sun = styled(SunIcon)`
  display: block;
  height: 0.5em;
  width: 0.5em;
  margin: 0 auto;
`;

/**
 * Stats Stateless Components.
 *
 * @param {Object} props - Component properties.
 * @returns {Object} ReactElement.
 */
export default function Stats({ value }) {
  return (
    <Container>
      <Sun fillColor="#FEC33A" />
      <PercentageLabel>
        {value}
        <span>
          %
        </span>
      </PercentageLabel>
      <BrightnessLabel>
        Brightness
      </BrightnessLabel>
    </Container>
  );
}

Stats.propTypes = {
  value: PropTypes.number.isRequired,
};
