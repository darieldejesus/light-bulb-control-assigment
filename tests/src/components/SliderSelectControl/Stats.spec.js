import React from 'react';

import Stats from '../../../../src/components/SliderSelectControl/Stats';

describe('Stats', () => {
  test('Should render the component', () => {
    const wrapper = shallow(<Stats value={45} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
