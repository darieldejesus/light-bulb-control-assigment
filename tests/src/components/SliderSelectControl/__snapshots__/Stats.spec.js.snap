// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Should render the component 1`] = `
<styled.div>
  <Styled(SunIcon)
    fillColor="#FEC33A"
  />
  <styled.span>
    45
    <span>
      %
    </span>
  </styled.span>
  <styled.p>
    Brightness
  </styled.p>
</styled.div>
`;
